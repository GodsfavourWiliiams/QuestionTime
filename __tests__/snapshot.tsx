import AuthenticationPage from '@/components/auth/Login';
import View from '@/components/pages/dashboard';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { render } from '@testing-library/react';

jest.mock('next/navigation', () => ({
  useRouter: () => ({
    push: jest.fn(),
    events: {
      on: jest.fn(),
      off: jest.fn(),
      emit: jest.fn(),
    },
    isFallback: false,
  }),
}));

it('renders auth unchanged', () => {
  const { container } = render(
    <QueryClientProvider client={new QueryClient()}>
      <AuthenticationPage />
    </QueryClientProvider>
  );
  expect(container).toMatchSnapshot();
});

it('renders dashboard unchanged', () => {
  const { container } = render(
    <QueryClientProvider client={new QueryClient()}>
      <View />
    </QueryClientProvider>
  );
  expect(container).toMatchSnapshot();
});