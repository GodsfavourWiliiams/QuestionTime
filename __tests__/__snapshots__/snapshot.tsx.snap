// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders auth unchanged 1`] = `
<div>
  <div
    class="w-full max-w-sm mx-auto h-screen flex-col items-center justify-center flex px-4 lg:px-0"
  >
    <div
      class="flex flex-col space-y-2 text-center"
    >
      <h1
        class="text-2xl font-semibold tracking-tight"
      >
        Question Time
      </h1>
      <p
        class="text-sm text-muted-foreground"
      >
        Access your QuestionTime account.
      </p>
    </div>
    <div
      class="grid gap-6 w-full mt-5"
    >
      <form
        class="grid gap-6"
      >
        <div
          class="space-y-2"
        >
          <label
            class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
            for="email"
          >
            Email Address
          </label>
          <div
            class="flex items-center h-max w-full rounded-xl border border-input bg-background text-base ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-grayscale-500 focus-within:outline-none focus-within:ring-2 focus-within:ring-primary focus-within:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          >
            <input
              aria-describedby=":r0:-form-item-description"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="email"
              autocorrect="off"
              class="flex-grow h-full w-full rounded-xl px-5 py-4 focus-within:outline-none focus-within:ring-transparent"
              id="email"
              name="email"
              placeholder="name@example.com"
              type="email"
              value=""
            />
          </div>
        </div>
        <button
          class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-11 rounded-[12px] px-8 py-7 w-full"
          type="submit"
        >
          Sign In with Email
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`renders dashboard unchanged 1`] = `
<div>
  <div
    class="app-container flex flex-col py-10 w-full gap-10"
  >
    <div
      class="flex flex-col gap-6 border py-4 md:py-6 bg-white border-[#F0F5F8] rounded-2xl w-full"
    >
      <div
        class="flex items-center justify-between text-primary w-full px-4 md:px-6"
      >
        <h1
          class="font-semibold text-xl sm:text-2xl"
        >
          All Questions
        </h1>
        <button
          class="justify-center whitespace-nowrap text-sm ring-offset-background transition-colors focus-visible:outline-none disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 rounded-[12px] w-fit font-semibold flex items-center gap-1.5 p-3 sm:p-6"
        >
          <svg
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M8 2.75C8 2.47386 7.77614 2.25 7.5 2.25C7.22386 2.25 7 2.47386 7 2.75V7H2.75C2.47386 7 2.25 7.22386 2.25 7.5C2.25 7.77614 2.47386 8 2.75 8H7V12.25C7 12.5261 7.22386 12.75 7.5 12.75C7.77614 12.75 8 12.5261 8 12.25V8H12.25C12.5261 8 12.75 7.77614 12.75 7.5C12.75 7.22386 12.5261 7 12.25 7H8V2.75Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <p
            class="hidden sm:block"
          >
            New Question
          </p>
        </button>
      </div>
      <div
        class="w-full "
      >
        <div
          class="rounded-md"
        >
          <div
            class="relative w-full overflow-auto"
          >
            <table
              class="w-full caption-bottom text-sm"
              data-testid="question-table"
            >
              <thead
                class="[&_tr]:border-b"
              >
                <tr
                  class="border-none transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted pl-3 bg-[#FAFAFA]"
                >
                  <th
                    class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0 py-4"
                  >
                    Question
                  </th>
                  <th
                    class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0 py-4"
                  >
                    Options
                  </th>
                  <th
                    class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0 py-4"
                  >
                    <span
                      class="w-full flex justify-end text-xs"
                    >
                      Actions
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody
                class="[&_tr:last-child]:border-0"
              >
                <tr
                  class="border-none transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                >
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                </tr>
                <tr
                  class="border-none transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                >
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                </tr>
                <tr
                  class="border-none transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                >
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                </tr>
                <tr
                  class="border-none transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                >
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                </tr>
                <tr
                  class="border-none transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                >
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                </tr>
                <tr
                  class="border-none transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                >
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                </tr>
                <tr
                  class="border-none transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                >
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                </tr>
                <tr
                  class="border-none transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                >
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                </tr>
                <tr
                  class="border-none transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                >
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                </tr>
                <tr
                  class="border-none transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                >
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0 py-6 pl-2 pr-3"
                  >
                    <div
                      class="w-full h-4 rounded-md animate-pulse bg-slate-200"
                    />
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
